src/components/PluginInstallationsChart.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useEffect, useRef } from 'react'
+import * as echarts from 'echarts'
+import dayjs from 'dayjs'
+import { PluginChartProps } from '../data/plugins'
+
+const PluginInstallationsChart: React.FC<PluginChartProps> = ({ data }) => {
+    const chartRef = useRef(null)
+
+    useEffect(() => {
```

Comment:
Think this could be either a useMemo or a useCallback, but not a useEffect, because we are not dealing with React side effects here: https://www.w3schools.com/react/react_useeffect.asp

Reply:
Items in top part of `useEffect` are not side effects I think, and should not be placed inside the `useEffect` hook.